src/main/java/org/frawa/elmtest/core/ElmProjectHelper.java

```diff
@@ -0,0 +1,42 @@
+package org.frawa.elmtest.core;
+
+import com.intellij.openapi.components.ServiceManager;
+import com.intellij.openapi.project.Project;
+import org.elm.workspace.ElmProject;
+import org.elm.workspace.ElmWorkspaceService;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+public class ElmProjectHelper {
+
+    private final ElmWorkspaceService workspaceService;
+
+    public ElmProjectHelper(Project project) {
+        workspaceService = ServiceManager.getService(project, ElmWorkspaceService.class);
+    }
+
+    public Stream<String> allNames() {
+        return workspaceService.getAllProjects().stream()
+                .filter(p -> Files.exists(p.getProjectDirPath().resolve("tests")))
+                .map(ElmProject::getPresentableName);
+    }
+
+    public Optional<String> nameByProjectDirPath(String path) {
+        return workspaceService.getAllProjects().stream()
+                .filter(p -> p.getProjectDirPath().toString().equals(path))
+                .map(ElmProject::getPresentableName)
+                .findFirst();
+    }
+
+    public Optional<String> projectDirPathByIndex(int index) {
+        List<ElmProject> allProjects = workspaceService.getAllProjects();
+        ElmProject project = 0 <= index && index < allProjects.size() ? allProjects.get(index) : null;
```

Comment:
This lookup is incorrect because the `index` is based on the filtered list which backs the combobox (excluding projects where the "tests" folder is absent).

Either it needs to operate on the filtered list or get rid of the filtering entirely and instead replace it with some kind of status/error icon when the user selects a project which is invalid. I think the latter would be best since it would give an opportunity to say _why_ it won't work.

Reply:
Indeed ... done.